babel/localedata.py

```diff
@@ -47,10 +51,11 @@ def exists(name):
     """
     if not name or not isinstance(name, string_types):
         return False
+
     if name in _cache:
         return True
-    file_found = os.path.exists(os.path.join(_dirname, '%s.dat' % name))
-    return True if file_found else bool(normalize_locale(name))
```

Comment:
The first condition is no longer needed, if the file exists on disk then `name` must be in `_locale_identifiers` and the call to `normalize_locale` will return a `str`.